### PR TITLE
Fix 8336 eslint path alias error

### PR DESCRIPTION
**Pull request by swalker326**
Closes: #8336 

Testing Strategy:

1. add a test component
1. triggered the error by importing it with the ~/
1. applied the fix to the .eslintrc
1. restarted eslint to make sure that:
  -the error was resolved on the import
  -eslint was running without error

### CODE REVIEW

File: templates/remix/app/components/Component.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export function Component() {
+  return (
+    <div>Am I real?</div>
+  )
+}
```
Comment (brophdawg11) [inline templates/remix/app/components/Component.tsx:5]: I confirmed this seems to work locally - mind removing this new component addition/usage before we get this merged?

**Comment by brophdawg11** [comment]
Thanks for the PR!  We'll also want to add this to all the other `templates/` (except `remix-tutorial` and `remix-javascript` I think) but I'm happy to do that part once this is ready for the primary template :)

File: templates/remix/package.json [hunk 2]
```diff
@@ -25,7 +25,8 @@
     "@typescript-eslint/eslint-plugin": "^6.7.4",
     "eslint": "^8.38.0",
     "eslint-config-prettier": "^9.0.0",
-    "eslint-plugin-import": "^2.28.1",
+    "eslint-import-resolver-typescript": "^3.6.1",
```
[the remark this answers is not in the document]
Comment (glitchassassin) [reply templates/remix/package.json:28]: On my end, it looks like the moved key is just causing eslint to crash before reporting any errors (other eslint errors are also suppressed)

File: templates/remix/app/components/Component.tsx (hunk 1 above)
Comment (swalker326) [reply templates/remix/app/components/Component.tsx:5]: Yea I was in the process of updating them when I realized something was still a bit off. Per [this comment](https://github.com/remix-run/remix/pull/8337#discussion_r1434506330)

I'm leaving the key in the correct place, and adding the package required for paths to work to each template. Tested it locally and it seems to work correctly. Sound good?

**Comment by swalker326** [comment]
So after much pain and troubleshooting I tracked the issue down to `"plugin:import/recommended"` Also chased down the issue in [their repo](https://github.com/import-js/eslint-plugin-import/issues/2765) and it looks like they maybe have a fix waiting deployment. Not sure if it will resolve the issue though. Removing the plugin allowed the project to lint, typecheck and run without issue. 

I also went into each template and did the following:

1. add a test component
2. triggered the error by importing it with the ~/ 
3. applied the fix to the .eslintrc
4. restarted eslint to make sure that: 
   - the error was resolved on the import 
   - eslint was running without error

**Comment by brophdawg11** [comment]
Hm, are we sure we want to remove the preset entirely?  Doesn't that remove other potentially valuable rules just to get rid of the one problematic one?  Personally, I'd prefer a targeted fix to the problematic rule.  Full disclosure - I don't know what else is in the preset - I'll try to poke in and check soon though

**Comment by brophdawg11** [comment]
I'm going to close this in favor of #8441 so we can keep the "recommended" preset and just fix the alias issue by adding the new `settings` config

**Pull request closed without merging**